sql: Check for concurrent DSC job in legacy schema changer

spilchen opened the pull request:
Running the legacy schema changer and the declarative schema changer concurrently can cause issues due to their different approaches to updating descriptors. Normally we have checks to prevent the legacy schema changer from running in such scenarios, timing issues persisted—particularly between `ALTER VIEW .. RENAME` (legacy) and `DROP VIEW` (DSC). In these cases, the view rename could delete the descriptor being processed by the drop view operation.

This change addresses the timing issue by adding a check for an active DSC job at the start of the legacy schema changer job. With this fix, the issue could no longer be reproduced, whereas it was consistently reproducible before.

Epic: none
Closes: #137487
Closes: https://github.com/cockroachdb/cockroach/issues/137828
Release note (bug fix): Fixed a timing issue between `ALTER VIEW .. RENAME` and `DROP VIEW` that caused repeated failures in the `DROP VIEW` job.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/137868)
<!-- Reviewable:end -->

rafiss reviewed (approved):
nice investigation!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/137868#-:-OE_wqhL4WRhpbbA7cqr:b8ayq8t)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

spilchen committed:
sql: Check for concurrent DSC job in legacy schema changer

Running the legacy schema changer and the declarative schema changer
concurrently can cause issues due to their different approaches to
updating descriptors. Normally we have checks to prevent the legacy
schema changer from running in such scenarios, timing issues
persisted—particularly between `ALTER VIEW .. RENAME` (legacy) and `DROP
VIEW` (DSC). In these cases, the view rename could delete the descriptor
being processed by the drop view operation.

This change addresses the timing issue by adding a check for an active
DSC job at the start of the legacy schema changer job. With this fix,
the issue could no longer be reproduced, whereas it was consistently
reproducible before.

Epic: none
Closes: #137487
Release note (bug fix): Fixed a timing issue between `ALTER VIEW ..
RENAME` and `DROP VIEW` that caused repeated failures in the `DROP VIEW`
job.
